ctk/common/src/main/kotlin/org/codice/compliance/verification/core/assertions/ConditionsVerifier.kt

```diff
@@ -24,78 +24,72 @@ import org.codice.compliance.SAMLCore_2_5_1_c
 import org.codice.compliance.allChildren
 import org.codice.compliance.children
 import org.codice.compliance.utils.TestCommon
+import org.codice.compliance.verification.core.CommonDataTypeVerifier
 import org.w3c.dom.Node
-import java.time.Instant
 
 internal class ConditionsVerifier(val node: Node) {
     companion object {
         private const val AUDIENCE = "Audience"
     }
 
+    /** 2.5 Conditions */
     fun verify() {
         node.allChildren("Conditions").forEach {
-            verifyConditionAttributes(it)
-            verifyConditionType(it)
-            verifyOneTimeUse(it)
+            verifyConditions(it)
+            verifyAudience(it)
             verifyProxyRestrictions(it)
         }
     }
 
     /**
-     * Verify the <Conditions> element against the Core Spec
      * 2.5.1 Element <Conditions>
      * 2.5.1.2 Attributes NotBefore and NotOnOrAfter
      * 2.5.1.5 Element <OneTimeUse>
-     * 2.5.1.6 Element <ProxyRestriction>
      */
-    private fun verifyConditionAttributes(conditionsElement: Node) {
+    private fun verifyConditions(conditionsElement: Node) {
         val notBefore = conditionsElement.attributes.getNamedItem("NotBefore")
         val notOnOrAfter = conditionsElement.attributes.getNamedItem("NotOnOrAfter")
-        if (notBefore != null
-                && notOnOrAfter != null) {
-            val notBeforeValue = Instant.parse(notBefore.textContent)
-            val notOnOrAfterValue = Instant.parse(notOnOrAfter.textContent)
-            if (notBeforeValue.isAfter(notOnOrAfterValue))
-                throw SAMLComplianceException.create(SAMLCore_2_5_1_2,
-                        message = "NotBefore element with value $notBeforeValue is not less than " +
-                                "NotOnOrAfter element with value $notOnOrAfterValue.",
-                        node = node)
-        }
-    }
 
-    private fun verifyConditionType(conditionsElement: Node) {
+        if (notBefore != null) CommonDataTypeVerifier.verifyDateTimeValues(notBefore)
+        if (notOnOrAfter != null) CommonDataTypeVerifier.verifyDateTimeValues(notOnOrAfter)
```

Comment:
❓ Should we move these dateTime validity checks within the validateTimeWindow method? Looks like all notBefore and notOnOrAfter attributes have to follow the 1.3.3 URI requirements.